packages/backend/src/tokens/request.ts

```diff
@@ -159,6 +178,20 @@ ${err.getFullMessage()}`,
     return signedIn(authenticateContext, verifyResult!, headers);
   }
 
+  function handleMaybeHandshakeStatus(
+    context: typeof authenticateContext,
+    reason: AuthErrorReason,
+    message: string,
+    headers?: Headers,
+  ) {
+    if (isRequestEligibleForHandshake(context)) {
+      // Right now the only usage of passing in different headers is for multi-domain sync, which redirects somewhere else.
+      // In the future if we want to decorate the handshake redirect with additional headers per call we need to tweak this logic.
+      return handshake(context, reason, message, headers ?? buildRedirectToHandshake());
+    }
+    return signedOut(context, reason, message, new Headers());
+  }
```

Comment:
This felt like a reasonable abstraction, easy to tweak the handshake heuristic in one place going forward.